files/bootstrap.sh

```diff
@@ -389,9 +405,43 @@ fi
 
 echo $B64_CLUSTER_CA | base64 -d > $CA_CERTIFICATE_FILE_PATH
 
-sed -i s,CLUSTER_NAME,$CLUSTER_NAME,g /var/lib/kubelet/kubeconfig
 sed -i s,MASTER_ENDPOINT,$APISERVER_ENDPOINT,g /var/lib/kubelet/kubeconfig
 sed -i s,AWS_REGION,$AWS_DEFAULT_REGION,g /var/lib/kubelet/kubeconfig
+
+if [[ -z "$ENABLE_LOCAL_OUTPOST" ]]; then
+    # Only when "--enable-local-outpost" option is not set explicity on calling bootstrap.sh, it will be assigned with 
+    #    - the result of auto-detectection through describe-cluster
+    #    - or "false" when describe-cluster is bypassed.
+    #  This also means if "--enable-local-outpost" option is set explicity, it will override auto-detection result
+    ENABLE_LOCAL_OUTPOST="${IS_LOCAL_OUTPOST_DETECTED:-false}"    
+fi
+
+### To support worker nodes to continue to communicate and connect to local cluster even when the Outpost 
+### is disconnected from the parent AWS Region, the following specific setup are required:
+###    - append entries to /etc/hosts with the mappings of control plane host IP address and API server 
+###      domain name. So that the domain name can be resolved to IP addresses locally.
+###    - use aws-iam-authenticator as bootstrap auth for kubelet TLS bootstrapping which downloads client 
+###      X.509 certificate and generate kubelet kubeconfig file which uses the cleint cert. So that the 
+###      worker node can be authentiacated through X.509 certificate which works for both connected and 
+####     disconnected state.
+if [[ "${ENABLE_LOCAL_OUTPOST}" == "true" ]]; then
+    ### append to /etc/hosts file with shuffled mappings of "IP address to API server domain name"
+    DOMAIN_NAME=$(echo "$APISERVER_ENDPOINT" | awk -F/ '{print $3}' | awk -F: '{print $1}')
+    getent hosts "$DOMAIN_NAME" | shuf >> /etc/hosts
+
+    ### kubelet bootstrap kubeconfig uses aws-iam-authenticator with cluster id to authenticate to cluster
+    ###   - if "aws eks describe-cluster" is bypassed, for local outpost, the value of CLUSTER_NAME parameter will be cluster id.
+    ###   - otherwise, the cluster id will use the id returned by "aws eks describe-cluster".
+    CLUSTER_ID="${CLUSTER_ID:-$CLUSTER_NAME}" 
+    sed -i s,CLUSTER_NAME,$CLUSTER_ID,g /var/lib/kubelet/kubeconfig
```

Comment:
I don't understand this. So if someone runs `/etc/eks/bootstrap.sh $clusterName`, then we will attempt to auto discover, and `$CLUSTER_ID` will be set. We'll wire in the clusterId rather than clusterName in kubeConfig. 

But if someone runs `/etc/eks/bootstrap.sh --api-server-endpoint-url foo --cluster-ca-data bar --enable-local-outpost true $clusterName` then we won't autoDiscover and `clusterId` won't be used in the kubeConfig at all. Is that correct?

Reply:
Yes. For the second scenario (support local outpost and bypass DescribeClustger), the cluster ID instead of clusterName should be passed to bootstrap.sh.